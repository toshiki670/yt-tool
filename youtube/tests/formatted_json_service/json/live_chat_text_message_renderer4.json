{
  "replayChatItemAction": {
    "actions": [
      {
        "addChatItemAction": {
          "item": {
            "liveChatTextMessageRenderer": {
              "message": {
                "runs": [
                  {
                    "text": "https://www.youtube.com/watch?v=CaBFk...",
                    "navigationEndpoint": {
                      "commandMetadata": {
                        "webCommandMetadata": {
                          "url": "https://www.youtube.com/watch?v=CaBFkMwhJ2w",
                          "webPageType": "WEB_PAGE_TYPE_UNKNOWN",
                          "rootVe": 83769
                        }
                      },
                      "urlEndpoint": {
                        "url": "https://www.youtube.com/watch?v=CaBFkMwhJ2w",
                        "target": "TARGET_NEW_WINDOW",
                        "nofollow": true
                      }
                    }
                  }
                ]
              },
              "authorName": { "simpleText": "投稿者名" },
              "authorPhoto": {
                "thumbnails": [
                  {
                    "url": "https://yt4.ggpht.com/ytc/AIdro_lTkRVt_td-dWTG1vhBTHHXbOYDkLSEeeb2cOaaQau-0A=s32-c-k-c0x00ffffff-no-rj",
                    "width": 32,
                    "height": 32
                  },
                  {
                    "url": "https://yt4.ggpht.com/ytc/AIdro_lTkRVt_td-dWTG1vhBTHHXbOYDkLSEeeb2cOaaQau-0A=s64-c-k-c0x00ffffff-no-rj",
                    "width": 64,
                    "height": 64
                  }
                ]
              },
              "contextMenuEndpoint": {
                "commandMetadata": {
                  "webCommandMetadata": { "ignoreNavigation": true }
                },
                "liveChatItemContextMenuEndpoint": {
                  "params": "Q2g0S0hBb2FRMUJRUXpKZk1tRnpORzlFUm1FM1FuZG5VV1JaYW1kNVlVRWFLU29uQ2hoVlF6WnZURGhuWlV0TlpYbG5iUzAwT1dsTWNuRmxabmNTQzNSQlFqa3dUVlZxY2xWVklBRW9CRElhQ2hoVlExVlJiR2d3TTNGR1UwbGthMVZSWVZweFpFVnBjbmM0QWtnQVVBRSUzRA=="
                }
              },
              "id": "ChwKGkNQUEMyXzJhczRvREZhN0J3Z1FkWWpneWFB",
              "timestampUsec": "1734589915000573",
              "authorBadges": [
                {
                  "liveChatAuthorBadgeRenderer": {
                    "icon": { "iconType": "MODERATOR" },
                    "tooltip": "Moderator",
                    "accessibility": {
                      "accessibilityData": { "label": "Moderator" }
                    }
                  }
                }
              ],
              "authorExternalChannelId": "UCUQlh03qFSIdkUQaZqdEirw",
              "contextMenuAccessibility": {
                "accessibilityData": { "label": "Chat actions" }
              }
            }
          },
          "clientId": "CPPC2_2as4oDFa7BwgQdYjgyaA"
        }
      }
    ]
  },
  "videoOffsetTimeMsec": "20787706",
  "isLive": true
}
