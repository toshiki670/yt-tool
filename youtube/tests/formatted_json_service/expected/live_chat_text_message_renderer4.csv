id,authorExternalChannelId,postedAt,category,authorName,content
ChwKGkNQUEMyXzJhczRvREZhN0J3Z1FkWWpneWFB,UCUQlh03qFSIdkUQaZqdEirw,2024-12-19T15:31:55.000573+09:00,Text message,投稿者名,message: 'https://www.youtube.com/watch?v=CaBFkMwhJ2w'; Moderator
